src/Renderer.cpp

```diff
@@ -0,0 +1,169 @@
+#include "Renderer.h"
+
+#include "AssetManager.h"
+
+#include <glm/gtc/matrix_transform.hpp>
+
+Renderer::Renderer(AssetManager& n_asset_manager)
+    : time(0.0f),
+      asset_manager(n_asset_manager) {
+    EventSystem::add_event_handler(EventType::LOAD_EVENT, &Renderer::load, this);
+    EventSystem::add_event_handler(EventType::KEYPRESS_EVENT, &Renderer::handle_key_press, this);
+}
+
+void Renderer::update() {
+    example_objects[1].apply_transform(glm::rotate(glm::mat4x4(), 0.01f, glm::vec3(1, 1, 1)));
+
+    //time += 0.01f;
+
+    //This should be moved to a different system
+    SDL_Event event;
+
+    while (SDL_PollEvent(&event)) {
+        switch (event.type) {
+            case SDL_QUIT:
+                std::cout << "SDL_QUIT was called" << std::endl;
```

Comment:
I would prefer that all quit logic be in one place. What I've done so far is defined a quit() function for the different systems and called those in the main loop. We can move all of them out altogether later on.